samcli/commands/build/build_context.py

```diff
@@ -415,8 +419,15 @@ def collect_all_build_resources(self) -> ResourcesToBuildCollector:
             ResourcesToBuildCollector that contains all the buildable resources.
         """
         result = ResourcesToBuildCollector()
-        result.add_functions([f for f in self.function_provider.get_all() if BuildContext._is_function_buildable(f)])
-        result.add_layers([l for l in self.layer_provider.get_all() if BuildContext._is_layer_buildable(l)])
+        excludes: List[str] = list(self._exclude) if self._exclude else []
```

Comment:
Originally I did this since you cannot iterate over `None` and `self._excludes` is an optional value, but using `Tuple` should be just as effective. I'll implement that in the following commit.